src/utils/types.ts

```diff
@@ -40,3 +40,7 @@ export type SuperiorPick<T, K extends keyof T> = T extends any ? {
  * Original Omit type extended to work on Union type objects
  */
 export type SuperiorOmit<T, K extends keyof any> = T extends any ? SuperiorPick<T, Exclude<keyof T, K>> : never;
+
+export type MakeUnion<Object extends object> = keyof Object extends infer Keys
```

Comment:
I am not quite sure I understand this type. I seems to make every property of the provided object required and remove `undefined` from its values. But that doesn't fit the name of the type so I am confused.

Reply:
it makes union type from a type of one object. Each item in union has one of the property so from {a: string, b: string} it would create {a: string} | {b:string}, I made that type as a return type to the function that checks whether the user gave the right parameters (name, all, next or range)

Reply:
Oh, I understand now. The new naming helps, but a small docs would help even more. Maybe with one example. 😉